### PR TITLE
Gather logs from previous iterations of restarted containers of non-e2e namespaced resources

### PR DESCRIPTION
**Pull request by rzetelskik**
**Description of your changes:** At the moment we are only gathering logs from previous iterations of restarted containers in e2e namespaces. This pull request introduces a modification that also gathers such logs from containers in other namespaced resources.

### CODE REVIEW

File: hack/ci-gather-artifacts.sh [hunk 1]
```diff
@@ -64,7 +64,12 @@ for n in ${namespaces}; do
                 containers=$( kubectl -n "${n}" get "pods/${o}" --template='{{ range .spec.containers }}{{ println .name }}{{ end }}' )
                 for c in ${containers}; do
                     # Logs are best effort, a pod might not have been running yet.
-                    { kubectl -n "${n}" logs pod/"${o}" -c="${c}" > "${c}".log; } || true
+                    { kubectl -n "${n}" logs pod/"${o}" -c="${c}" > "${c}".current; } || true
+
+                    previous=$( kubectl -n "${n}" get "pods/${o}" --template="{{ range .status.containerStatuses }}{{ if eq .name \"${c}\" }}{{ gt .restartCount 0 | print }}{{ end }}{{ end }}" )
```
Comment (tnozicka) [inline hack/ci-gather-artifacts.sh:69]:
```suggestion
                    has_previous=$( kubectl -n "${n}" get "pods/${o}" --template="{{ range .status.containerStatuses }}{{ if eq .name \"${c}\" }}{{ gt .restartCount 0 | print }}{{ end }}{{ end }}" )
```

File: hack/ci-gather-artifacts.sh [hunk 2]
```diff
@@ -64,7 +64,12 @@ for n in ${namespaces}; do
...
+                    if [[ "${previous}" = true ]]; then
```
Comment (tnozicka) [inline hack/ci-gather-artifacts.sh:70]: nit: the second value should be quoted, but would it be more defensive to check just for non empty string like with grep?

Comment (rzetelskik) [reply hack/ci-gather-artifacts.sh:70]: previous evaluates to either true or false, why should it check for a non-empty string?

Comment (tnozicka) [reply hack/ci-gather-artifacts.sh]: both are fine. the idea was to print a string only if greater then zero and then you don't have to care about the value, it's format (e.g. if a next version prints it with quotes or a capital letter) or if there are more then one match (although container names have to be unique)

**Review by tnozicka — APPROVED** [review]
/approve
/lgtm

**Pull request merged**